lua/toggleterm/terminal.lua

```diff
@@ -132,8 +132,20 @@ end
 --- @param bufnr number
 local function setup_buffer_mappings(bufnr)
   local mapping = config.open_mapping
-  if mapping and config.terminal_mappings then
-    vim.keymap.set("t", mapping, "<Cmd>ToggleTerm<CR>", { buffer = bufnr, silent = true })
+
+  -- Key mapping function
+  ---@param key string
+  local map = function(key)
+    vim.keymap.set("t", key, "<Cmd>ToggleTerm<CR>", { buffer = bufnr, silent = true })
+  end
+
+  if config.terminal_mappings then
+    if type(mapping) == "string" then map(mapping) end
+    if type(mapping) == "table" then
+      for _, key in pairs(mapping) do
+        map(key)
+      end
+    end
```

Comment:
Looks to me like this logic is just being repeated maybe it's best to do this once in `utils.lua` generically and use that helper function in both of the places you've made this change